src/plugins/data/common/search/expressions/esql.ts

```diff
@@ -243,15 +243,34 @@ export const getEsqlFn = ({ getStartDependencies }: EsqlFnArguments) => {
               name,
               meta: { type: normalizeType(type) },
             })) ?? [];
-          const columnNames = columns.map(({ name }) => name);
+
+          // all_columns in the response means that there is a separation between
+          // columns with data and empty columns
+          // columns contain only columns with data while all_columns everything
+          const difference =
+            body.all_columns?.filter((col1) => {
+              return !body.columns.some((col2) => {
+                return col1.name === col2.name;
+              });
+            }) ?? [];
+          const emptyColumns =
+            difference?.map(({ name, type }) => ({
+              id: name,
+              name,
+              meta: { type: normalizeType(type) },
+              isNull: true,
+            })) ?? [];
+
+          const allColumns = [...columns, ...emptyColumns];
```

Comment:
We could optimize a little by checking first that `body.all_columns.length > body.columns.length`